### ISSUE TITLE
How to see how a View is structured and copy that view to new database

### ISSUE THREAD

**Issue by vayurobins** [TablePlus/TablePlus#2186]
1. Microsft SQL Server connection
2. Version 3.10.0 (348)

How can I see how a View is structured and how can copy that View query and use the same View on another Table?

**Comment by huyphams** [comment]
Hi @vayurobins there is a structure button below the view content. If you have any questions please email nick@tableplus.com we will reply to you faster!
Thank you!

**Comment by vayurobins** [comment]
Thank you @huyphams ! Yes, I did have plenty looks into that before posting, but it makes no sense and it is 1000 lines long. So obviously it displays some information about what is happening, but it is not really useful. How could I copy the View and paste it elsewhere? Or how could export and import it?

**Comment by huyphams** [comment]
TablePlus didn't generate view definition on its own. It was returned from the database. So i think there are some view mode. Let me check it!

**Comment by vayurobins** [comment]
Thanks @huyphams, I look forward to hearing from you, and hope that you might find a solution :-)

**Comment by vayurobins** [comment]
Just to add extra info, in my Structure there is nothing, it's empty. I can only see history. I've attached a screenshot of the View structure
![Screenshot 2020-11-17 at 16 19 30](https://user-images.githubusercontent.com/1386354/99408587-b0e51f80-28f0-11eb-8f80-b2d5f668c569.png)

**Comment by huyphams** [comment]
Hi @vayurobins TablePlus uses this query to get the definition

SELECT m.definition AS create_function FROM sys.sql_modules m INNER JOIN sys.objects o ON m.object_id=o.object_id INNER JOIN sys.schemas s ON o.schema_id=s.schema_id WHERE s.name='schema_name' AND o.name='function_name';

Please repace  schema_name and function_name and let me know the result.

**Comment by vayurobins** [comment]
Thanks! :-) I get this error: ```Table 'sys.sql_modules' doesn't exist```

**Comment by vayurobins** [comment]
How come the Structure window is blank on the staging server? Is there anything else I can do to copy a View table to my local setup?

**Comment by huyphams** [comment]
Which version of SQL Server are you using @vayurobins?

**Comment by vayurobins** [comment]
Thanks for asking. It is version 10.0.17763 Build 17763

**Comment by huyphams** [comment]
It looks like a Windows version to me @vayurobins are you using SQL Server 2000?

**Comment by vayurobins** [comment]
Oh sorry, yes, Version 18.6

**Comment by huyphams** [comment]
The 18.6 is the version of the SQL Studio, could you please copy the SQL Sever version on the TablePlus Status bar (in case you don't know the version)

**Comment by vayurobins** [comment]
Hmm sorry, I am not sure. I can see when I click on the status bar in TablePlus, it reads Driver: Microsoft SQL Server 2019. I'm not sure if this is what you are asking for. Is that useful?

**Comment by huyphams** [comment]
Yes, that is the one we need, I need to know the version so I can download that version and try to reproduce the issue, there are various version of SQL Server

**Comment by vayurobins** [comment]
Great, thanks for looking into this, I appreciate it! :-) Let me know if there is anything else I can inform you of.

**Comment by vayurobins** [comment]
Hi @huyphams 

Did you ever find any solution to my issue?
